onnx_tf/opr_checker.py

```diff
@@ -0,0 +1,133 @@
+import argparse
+import warnings
+import logging
+import os
+
+import tensorflow as tf
+from google.protobuf import text_format
+
+from onnx import defs
+from onnx_tf import opset_version
+from onnx_tf.common import get_output_node_names
+from onnx_tf.common.handler_helper import get_all_frontend_handlers
+from onnx_tf.pb_wrapper import TensorflowNode
+
+warnings.filterwarnings('ignore')
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger()
+
+
+def parse_args(args):
+  parser = argparse.ArgumentParser(
+      description=
+      'This is the checker to check whether model graph '\
+      'operators are supported by the frontend handlers'
+  )
+  parser.add_argument(
+      "--infile",
+      "-i",
+      help="Input file path, can be pb, pbtxt, or ckpt file.",
+      required=True)
+
+  return parser.parse_args(args)
+
+
+def check_opr_support(graph_def):
+  logger.info('Checking for unsupported operators...')
+  node_dict = set()
+  unsupported = set()
+  supported = set()
+  for node in graph_def.node:
+    node_dict.add(node.op)
+
+  node_dict.discard('Placeholder')
+  node_dict.discard('Const')
+
+  logger.info('There are %s unique operators in the model file.',
+              str(len(node_dict)))
+  for k in node_dict:
+    if k not in opset_version.frontend_tf_opset_version:
+      unsupported.add(k)
+    else:
+      supported.add(k)
+
+  if unsupported:
+    logger.info(
+        'There are %s operators currently not supported by the ONNX '\
```

Comment:
`not supported by the ONNX` is not precise.

Reply:
The sentence is precise "There are %s operators currently not supported by the ONNX exporter for your model." The supported tf operators are extracted from the frontend handlers' @tf_op("xxx"). If an operator is not declared in any handlers, it is not currently supported by the exporter/frontend. If the comment is about the wording "ONNX exporter" I can change it to "ONNX Tensorflow frontend handler".

Reply:
ONNX-Tensorflow should be better if you want to use full name.
Seems strange without hyphen...

Reply:
Just added hyphen